### PR TITLE
[stable6] Backport of stable 6 for Nextcloud 26

### PR DESCRIPTION
**Pull request by come-nc**
1. Fixed signature of `beforeController` method on the middleware
2. Allow 26 in info.xml
3. Replace attributes by annotations

This PR also includes the following PRs:
- https://github.com/nextcloud/polls/pull/3445
- https://github.com/nextcloud/polls/pull/3446
- https://github.com/nextcloud/polls/pull/3449
- https://github.com/nextcloud/polls/pull/3448
- https://github.com/nextcloud/polls/pull/3451

### CODE REVIEW

**Comment by dartcafe** [comment]
nice. Now I remember the incompatibility. 😁

**Comment by dartcafe** [comment]
I updated the branch protection for stable-6

**Comment by come-nc** [comment]
> nice. Now I remember the incompatibility. 😁

Do you think I missed anything? Do you know how complete are the unit tests?
I tested the UI a bit and it seems to work, but there is always the possibility of an incompatible signature in a corner-case code path.
Psalm seems happy so that’s good.

**Comment by dartcafe** [comment]
As soon as you are ready, add me as a reviewer. I will check then. Code wise it looks good.

**Comment by dartcafe** [comment]
I installed NC26 and will test some recent edge cases. But it may make sense to add the performance issues to the branch, so the stable-5 can die.

File: .github/workflows/static-analysis.yml [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-      - master
-      - next
-      - stable*
-
-jobs:
-  psalm-master:
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        ocp-version: ['master']
```
Comment (artonge) [inline .github/workflows/static-analysis.yml:16]: Shouldn't we also test for upper versions?

File: lib/Controller/VoteController.php [hunk 2]
```diff
@@ -44,34 +42,34 @@ public function __construct(
 	}
 
 	/**
-	 * list votes per poll
+	 * set
```
Comment (artonge) [inline lib/Controller/VoteController.php:45]: You lost the comment

File: lib/Controller/VoteController.php [hunk 3]
```diff
@@ -44,34 +42,34 @@ public function __construct(
...
 	 */
-	#[NoAdminRequired]
-	#[NoCSRFRequired]
 	public function list(int $pollId): JSONResponse {
 		return $this->response(fn () => ['votes' => $this->voteService->list($pollId)]);
 	}
 
 	/**
-	 * set vote answer
+	 * set
```
Comment (artonge) [inline lib/Controller/VoteController.php:54]: Same, comment

**Review by artonge — DISMISSED** [review]

**Review by artonge — DISMISSED** [review]

**Comment by dartcafe** [comment]
I did some skoke testing and did not find any flaws so far. So OK from me!

> Do you know how complete are the unit tests?

The whole testing is very poor and would need some significant coverage increase. See #3147 😉

**Review by dartcafe — DISMISSED** [review]

**Comment by come-nc** [comment]
- [ ] https://github.com/nextcloud/polls/pull/3451 should be reverted
- [ ] `master` should be added back to phpunit and maybe psalm workflows
- [ ] Then merge :tada:

**Comment by dartcafe** [comment]
- https://github.com/nextcloud/polls/pull/3451 should be reverted 
  - see #3468 
- `master` should be added back to phpunit and maybe psalm workflows
  - Master is fine as it is, as well as the stable-6, or what do you have to do there?
- Then merge :tada: 
  - Please merge, when ever you are ready to (also to stable-6 ??)

**Comment by dartcafe** [comment]
What's about the open PRs? Aren't they superseeded and are to be closed? 
* #3380 
* #3368 
* #3364 
* #3363

**Comment by come-nc** [comment]
> * Please merge, whenever you are ready to (also to stable-6 ??)

I’m ready to merge, I just need your approval.

**Comment by dartcafe** [comment]
Approval already given. What do you think about merging #3467 first to this branch? 
It will help to reduce the poll loading time by half.

**Review by dartcafe — APPROVED** [review]

**Comment by come-nc** [comment]
> What do you think about merging #3467 first to this branch?
> It will help to reduce the poll loading time by half.

I think it’s cleaner to merge to stable-6.
This branch was meant as a temporary effort to test if supporting 26 was possible, now that it’s merged we can merge other things to stable-6 to keep improvements going.
